plugin/vim-gh-line.vim

```diff
@@ -48,14 +48,27 @@ if !exists('g:gh_cgit_url_pattern_sub')
     let g:gh_cgit_url_pattern_sub = []
 endif
 
-func! s:gh_line(action) range
+if !exists('g:git_remote')
+    let g:git_remote = ""
+endif
+
+func! s:gh_line(action, force_interactive) range
     " Get Line Number/s
     let lineNum = line('.')
     let fileName = resolve(expand('%:t'))
     let fileDir = resolve(expand("%:p:h"))
     let cdDir = "cd '" . fileDir . "'; "
 
-    let origin = system(cdDir . "git config --get remote.origin.url")
+    " try to find git remote
+    if a:force_interactive == 1 || g:git_remote == ""
```

Comment:
If there is only one remote, the plugin should just work, in all possible combinations. 

In this implementation, if force_interactive is not used and there is only a single remote, the plugin still expects the remote to be mentioned in the g:gh_git_remote variable. This is unoptimal.

Reply:
Can we parse the remotes here first and if there is only one remote, directly use that without considering a:force_interactive or g:gh_git_remote or calling s:find_git_remote ?

Reply:
Can we also keep the heuristic of using origin if force_interactive is not passed? That was working in most of the use cases and there should not be a reason to remove that heuristic.
If that heuristic does not work for a user, she can call :GHInteractive instead of using the 'gh' key combination.

Reply:
> If there is only one remote, the plugin should just work, in all possible combinations.
> 
> In this implementation, if force_interactive is not used and there is only a single remote, the plugin still expects the remote to be mentioned in the g:gh_git_remote variable. This is unoptimal.

what do you mean `force_interactive` is not used? `g:git_remote` will always be used to store the last used remote. I don't understand why it's unoptimal? can you be more specific?

Reply:
> Can we also keep the heuristic of using origin if force_interactive is not passed? That was working in most of the use cases and there should not be a reason to remove that heuristic.
> If that heuristic does not work for a user, she can call :GHInteractive instead of using the 'gh' key combination.

If you check the implementation of `s:find_git_remote`, the difference here by comparing with previous implementation in the case of single remote is we no longer expect `origin` as remote name, if you have single remote, it will work just as before(no user input needed), but it will handle any single remote name, not only `origin`.

Reply:
I see. 
I read the condition 
`if a:force_interactive == 1 || g:git_remote == ""` 
wrong. 

g:git_remote looks like an implementation variable rather than a global configuration variable to be set by users. We should probably not document it. 

Instead of users setting it from .vimrc, probably they should run '<Leader>gh' once and update g:git_remote through that. That way each "session" can have a different value in `g:git_remote`. 

This makes sense to me now. Thanks for clarifications.

Reply:
`g:git_remote` got me confused as well. May I suggest a different name for this variable, such as `g:gh_last_git_remote`?

Reply:
@KostyaEsmukov sounds good :)